lib/nats.js

```diff
@@ -66,26 +66,69 @@ var VERSION = '0.6.8',
     PONG_RESPONSE = 'PONG' + CR_LF,
 
     // Errors
-    BAD_SUBJECT = 'Subject must be supplied',
-    BAD_MSG = 'Message can\'t be a function',
-    BAD_REPLY = 'Reply can\'t be a function',
-    CONN_CLOSED = 'Connection closed',
+    BAD_SUBJECT = 'BAD_SUBJECT',
+    BAD_SUBJECT_MSG = 'Subject must be supplied',
+    BAD_MSG = 'BAD_MSG',
+    BAD_MSG_MSG = 'Message can\'t be a function',
+    BAD_REPLY = 'BAD_REPLY',
+    BAD_REPLY_MSG = 'Reply can\'t be a function',
+    CONN_CLOSED = 'CONN_CLOSED',
+    CONN_CLOSED_MSG = 'Connection closed',
+    BAD_JSON = 'BAD_JSON',
     BAD_JSON_MSG = 'Message should be a JSON object',
-    BAD_AUTHENTICATION = 'User and Token can not both be provided',
+    BAD_AUTHENTICATION = 'BAD_AUTHENTICATION',
+    BAD_AUTHENTICATION_MSG = 'User and Token can not both be provided',
+    INVALID_ENCODING = 'INVALID_ENCODING',
+    INVALID_ENCODING_MSG_PREFIX = 'Invalid Encoding:',
+    SECURE_CONN_REQ = 'SECURE_CONN_REQ',
+    SECURE_CONN_REQ_MSG = 'Server requires a secure connection.',
+    NON_SECURE_CONN_REQ = 'NON_SECURE_CONN_REQ',
+    NON_SECURE_CONN_REQ_MSG = 'Server does not support a secure connection.',
+    CLIENT_CERT_REQ = 'CLIENT_CERT_REQ',
+    CLIENT_CERT_REQ_MSG = 'Server requires a client certificate.',
+    CONN_ERR = 'CONN_ERR',
+    CONN_ERR_MSG_PREFIX = 'Could not connect to server: ',
+    NATS_PROTOCOL_ERR = 'NATS_PROTOCOL_ERR',
 
     // Pedantic Mode support
     //Q_SUB = /^([^\.\*>\s]+|>$|\*)(\.([^\.\*>\s]+|>$|\*))*$/, // TODO: remove / never used
     //Q_SUB_NO_WC = /^([^\.\*>\s]+)(\.([^\.\*>\s]+))*$/, // TODO: remove / never used
 
     FLUSH_THRESHOLD = 65536;
 
+
+  var NatsError = function(message, code, chainedError) {
+    Error.captureStackTrace(this, this.constructor);
+    this.name = this.constructor.name;
```

Comment:
ah, I think for `this.constructor.name` to be set here you have to declare it like ` var NatsError = function NatsError(err){`, as it right now might just be empty